### DISCUSSION TITLE
Calculate the number of pixels

### DISCUSSION THREAD

**Discussion by sihj2002** [dlemstra/Magick.NET#627] [General] [answered]
I want to calculate the number of black pixels.

![magick question 2](https://user-images.githubusercontent.com/64253749/81373487-d28e6e80-9137-11ea-90a5-e01085fe571c.jpg)

And this is my test code.

            string filepath = "C:/....tif";
            MagickImage img = new MagickImage(filepath);
                        
            img.ColorType = ColorType.Bilevel;

            img.Depth = 1;

            IPixelCollection pixels = img.GetPixels();
                       
            for (int x = 0; x < img.Width; x++)
            {
                for (int y = 0; y < img.Height; y++)
                {
                    Console.WriteLine(pixels.GetValue (x,y));
                    Console.WriteLine(y);
                }
             }

And this is the result.

![magick question](https://user-images.githubusercontent.com/64253749/81373193-1cc32000-9137-11ea-936d-8972c2601622.jpg)

I don't know how can I receive the pixel values in integer.
Or is there another method approaching to pixel?
Can you recommend it?

**Comment by dlemstra** [comment]
Can you share the original tiff image?

**Reply by sihj2002** [reply]
https://drive.google.com/open?id=1Kyoc_GGwceS6b70TyA1YrpVGVTOWt2TI

Here is tiff image.

**Comment by dlemstra** [comment, marked as answer]
For this particular image you could use the `Histogram` method of `MagickImage`:

```C#
using (var image = new MagickImage("test.tif"))
{
    var colors = image.Histogram();
    var blackPixelCount = colors[MagickColors.Black];
}
```

**Reply by sihj2002** [reply]
okay thanks I'll try and see.

**Reply by sihj2002** [reply]
It looks like it's working. I tried to calculate pixels of full black image and result is good.
test_1 's width and height is 11811 x 9449.

![magick question](https://user-images.githubusercontent.com/64253749/81529395-bb4ebb80-9399-11ea-84c6-db813b8d6204.jpg)

Thank you very much.
